test/raven_command_tests/QueryOperationsTest.ts

```diff
@@ -50,25 +49,21 @@ describe('IndexBasedCommand tests', () => {
 
   describe('Actions by Index', () => {
     it('update by index success', async () => {
-      const query: string = "from index 'Testing_Sort' where exists(Name)";
-      const indexQuery: IndexQuery = new IndexQuery(query, null, 0, {}, {waitForNonStaleResults: true, waitForNonStaleResultsAsOfNow: true});
-      const queryCommand: QueryCommand = new QueryCommand(indexQuery, new DocumentConventions());
-      const patchByQueryOperation: PatchByQueryOperation = new PatchByQueryOperation(new IndexQuery(query), patch, new QueryOperationOptions(false));
+      const query: string = "from index 'Testing_Sort' where exists(Name) update { this.Name = args.name; }";
+      const indexQuery: IndexQuery = new IndexQuery(query, {name: 'Patched'}, null, 0, {waitForNonStaleResults: true});
+      const patchByQueryOperation: PatchByQueryOperation = new PatchByQueryOperation(indexQuery, new QueryOperationOptions(false));
 
-      return requestExecutor
-        .execute(queryCommand)
-        .then(() => store.operations
-          .send(patchByQueryOperation)
-          .then((response: IRavenResponse) => {
-            expect(response).not.to.be.null;
-            expect((response as IRavenResponse).Result.Total).not.to.be.lessThan(100);
-          })
-        );
+      return store.operations
+        .send(patchByQueryOperation)
+        .then((response: IRavenResponse) => {
+          expect(response).not.to.be.null;
+          expect((response as IRavenResponse).Result.Total).not.to.be.lessThan(100);
+        });
     });
 
     it('update by index fail', async () => expect(
       store.operations
-        .send(new PatchByQueryOperation(new IndexQuery(`from index 'unexisting_index_1' where Name = 'test1'`), patch))
+        .send(new PatchByQueryOperation(new IndexQuery(`from index 'unexisting_index_1' where Name = 'test1' update { this.Name = args.name; }`, {name: 'Patched'})))
       ).to.be.rejected
```

Comment:
Can we be more specific on what error we actually expect using `chai`'s `promise.should.be.rejectedWith(Error)` call or other?